Bouncer/Views/FilterDetail/FilterDetailView.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  FilterDetailView.swift
+//  Bouncer
+//
+
+import SwiftUI
+
+enum InteractionType: Equatable {
+    case add
+    case update
+}
+
+struct FilterDetailView: View {
+    
+    var onSave: (Filter) -> Void
+    var interactionType: InteractionType
+    var filter: Filter?
+    
+    @Environment(\.presentationMode) var presentationMode
+    @State var filterType: FilterType
+    @State var filterDestination: FilterDestination
+    @State var filterTerm: String
+    @State var exactMatch: Bool
+    
+    var body: some View {
+        switch interactionType {
```

Comment:
Let's move this logic to the Container View.   We can pass these strings and buttons as part of the view initialization, or we can have this view call a method in the container view to get them.   

We should do as much as possible to remove any logic from the final view.

Reply:
You just made me change the whole approach with these Container and Composition patterns, so stay still, I'm working on it but is taking a bit longer than expected.

Reply:
@afterxleep this is the best I could do so far with my basic SwiftUI understanding. I think you are not gonna like it because of the validation on the **rootView** computed property but I just didn't find a better way to pass the navigationBarItems to this view. PAIN.